sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -66,29 +51,36 @@ class AzureAppConfigurationClient:
     def __init__(self, base_url: str, credential: TokenCredential, **kwargs: Any) -> None:
         try:
             if not base_url.lower().startswith("http"):
-                base_url = "https://" + base_url
+                base_url = f"https://{base_url}"
         except AttributeError as exc:
             raise ValueError("Base URL must be a string.") from exc
 
         if not credential:
             raise ValueError("Missing credential")
 
-        self._credential_scopes = base_url.strip("/") + "/.default"
-
-        self._config = AzureAppConfigurationConfiguration(base_url, credential_scopes=self._credential_scopes, **kwargs)
-        self._config.user_agent_policy = UserAgentPolicy(base_user_agent=USER_AGENT, **kwargs)
+        credential_scopes = [f"{base_url.strip('/')}/.default"]
         self._sync_token_policy = SyncTokenPolicy()
 
-        pipeline = kwargs.get("pipeline")
-
-        if pipeline is None:
-            aad_mode = not isinstance(credential, AppConfigConnectionStringCredential)
-            pipeline = self._create_appconfig_pipeline(
-                credential=credential, credential_scopes=self._credential_scopes, aad_mode=aad_mode, **kwargs
+        if isinstance(credential, AzureKeyCredential):
+            id_credential = kwargs.pop("id_credential")
+            kwargs.update(
+                {
+                    "authentication_policy": AppConfigRequestsCredentialsPolicy(credential, base_url, id_credential),
+                }
             )
-
+        elif isinstance(credential, TokenCredential):
+            kwargs.update(
+                {
+                    "authentication_policy": BearerTokenCredentialPolicy(credential, *credential_scopes, **kwargs),
+                }
+            )
+        else:
+            raise TypeError(
+                f"Unsupported credential: {type(credential)}. Use an instance of token credential from azure.identity"
```

Comment:
Or `AzureKeyCredential`?

Reply:
`AzureKeyCredential` is not for public use, add it may confuse customers

Reply:
then `AppConfigConnectionStringCredential`?

Reply:
`AzureKeyCredential` is for connection string, we no longer need the custom credential `AppConfigConnectionStringCredential`

Reply:
It is in our public surface, no?

Reply:
Only `Async/TokenCredential` in public surface.

Reply:
I saw 
```py
:type credential: ~azure.appconfiguration.AppConfigConnectionStringCredential
        or ~azure.core.credentials.TokenCredential
```
?

Reply:
Are you checking the latest updates?

Reply:
It shows in your PR's diff. :)

Reply:
Ah, the docstring is not consistent with the type hint. And `AppConfigConnectionStringCredential` is not a public type.